module/apmhttp/clienttrace.go

```diff
@@ -60,7 +60,7 @@ func withClientTrace(ctx context.Context, tx *apm.Transaction, parent *apm.Span)
 			r.mu.Lock()
 			defer r.mu.Unlock()
 			if !r.ended {
-				r.DNS = tx.StartSpan(fmt.Sprintf("DNS %s", i.Host), "external.http.dns", parent)
+				r.DNS = tx.StartExitSpan(fmt.Sprintf("DNS %s", i.Host), "external.http.dns", parent)
```

Comment:
The spec says:

> Exit spans MAY have child spans that have the same type and subtype. For example, an HTTP exit span may have child spans with the action request, response, connect, dns. These spans MUST NOT have any destination context, so that there's no effect on destination metrics.

If they mustn't have destination context, does that mean they should not be exit spans?

Reply:
It is not specified but I think it makes sense for them not to be exit spans, especially considering the related features (compression, etc.)